Debug gesture and logout OAuth endpoint

adamhrz opened the pull request:
Adds a two finger triple tap gesture to show the debug menu: 
* Shows build number
* Simulates OAuth bad refresh token
* Simulates OAuth bad access token
* Can be used to force a logout

KingOfBrian commented on PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift:
```diff
@@ -26,7 +26,83 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
+        // Don't load the main UI if we're unit testing.
+        if let _: AnyClass = NSClassFromString("XCTest") {
         return true
     }
```
I think the indentation is off here, and I'd prefer personally to run this before the configurations ran. Often tests want to run without the app configurations running.

KingOfBrian commented on PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift:
```diff
@@ -26,7 +26,83 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
+        window = UIWindow(frame: UIScreen.main.bounds)
+
+        // Adds (by default) a 2 finger triple tap gesture to present a debug menu
+        enableDebugGesture()
```
Actually, can we move this entire behavior into an `AppLifecycle `? It'd be encapsulated nicely and thin out the AppDelegate.

adamhrz replied:
👍

KingOfBrian commented on PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift:
```diff
@@ -26,7 +26,83 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
+            w.backgroundColor = UIColor.white
+
+            // Setting a tintcolor on the window adds a tint to all UIAlertControllers
+            w.tintColor = UIColor.blue
```
I'd prefer to leave styling out for the template, but not a big deal

adamhrz replied:
My thought was something to suggest where app hide window default would happen - happy to take it out

adamhrz commented:
@KingOfBrian This is ready for another look. I addressed your concerns moving the debug menu a config, removed the placeholder empty VC and re-generated the cookie cutter template

KingOfBrian commented on PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift:
```diff
@@ -18,15 +18,36 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
+
+        window = UIWindow(frame: UIScreen.main.bounds)
+        showTabBar(animated: false)
+        window?.makeKeyAndVisible()
```
Can we move `showTabBar` and the `makeKeyAndVisible` to after the config?

KingOfBrian replied:
Also, maybe make `showTabBar` more generic, like `configureRootViewController`?
